### DISCUSSION TITLE
Lesson 6: Failed Metamask to Ganache connection

### DISCUSSION THREAD

**Discussion by Renzo1** [Cyfrin/foundry-full-course-cu#258] [Q&A] [answered]
I am getting an error trying to add my Local Ganache network to metamask custom netrworks.

All the entry fields work correctly but Chain ID.

I am absolutely sure my Chain ID is correct because I have crosschecked it multiple times. I have even restart my ganache with new networks, and it's still not working.

I am not running VS code on WSL and the setting of Genache Server HOSTNAME is set to ```0.0.0.0 - All interfaces```.

Metamask keeps returning the below error for all my attempts.

```Could not fetch chain ID. Is your RPC URL correct?```

Please I need assistance in fixing this.

**Comment by shikhar229169** [comment]
Set chainid to 1337 in ganache. It worked for me.
Also, don't forget to set this chainid also in metamask for ganache.

If this doesn't work, then add the vEthernet wsl rpc url in metamask.

**Reply by Renzo1** [reply, marked as answer]
It's interesting because I followed all these steps, but unfortunately, it didn't work for me. However, I discovered a workaround using Anvil. I filled in the input in the MetaMask fields without clicking submit, then switched back to the Ganache input. Additionally, I had to modify the last part of my URL from 8545 to 7545 to avoid conflicts with Anvil. With a little tinkering, some frustration, I managed to get through it.
